AeonCore/main.py

```
@@ -0,0 +1,514 @@
+            ded = gradient_s + alpha * ric_t + beta * psi_t + gamma * intent_effect
+            ded_values.append(ded + np.random.normal(0, noise_term))
+        return ded_values
+
+    def phase_lock_intent(self, state: np.ndarray, intent: np.ndarray, 
+                         freq1: float = 117, period2: float = 7.83, 
+                         beta_base: float = 0.4) -> tuple[np.ndarray, List[str]]:
+        """Phase-lock intent to 117 Hz and 7.83 s signals."""
+        t = np.linspace(0, period2, 100)
+        f2 = 1 / period2
+        phi = 0.0
+        signal = np.sin(2 * np.pi * f1 * t + phi) + 0.1 * np.sin(2 * np.pi * f2 * t)
```

Ruff (0.12.2)

Undefined name `f1`

(F821)